Lombiq.Tests.UI/SecurityScanning/SecurityScanConfiguration.cs

```diff
@@ -0,0 +1,216 @@
+/// </summary>
+/// <remarks>
+/// <para>
+/// This class and <see cref="YamlDocumentExtensions"/> intentionally use different terminology, the latter assuming you
+/// know ZAP. Here, we provide a simplified configuration for people who just want to use security scans without having
+/// to understand ZAP's configuration too much.
+/// </para>
+/// </remarks>
+public class SecurityScanConfiguration
+{
+    public Uri StartUri { get; private set; }
+    public bool AjaxSpiderIsUsed { get; private set; }
+    public string SignInUserName { get; private set; }
+    public IList<string> ExcludedUrlRegexPatterns { get; } = new List<string>();
+    public IList<ScanRule> DisabledActiveScanRules { get; } = new List<ScanRule>();
+    public IList<ScanRule> DisabledPassiveScanRules { get; } = new List<ScanRule>();
+    public IDictionary<string, ScanRule> DisabledRulesForUrls { get; } = new Dictionary<string, ScanRule>();
+    public IList<Func<YamlDocument, Task>> ZapPlanModifiers { get; } = new List<Func<YamlDocument, Task>>();
+
+    internal SecurityScanConfiguration()
+    {
+    }
+
+    /// <summary>
+    /// Sets the start URL under the app where to start the scan from.
+    /// </summary>
+    /// <param name="startUri">The <see cref="Uri"/> under the app where to start the scan from.</param>
+    public SecurityScanConfiguration StartAtUri(Uri startUri)
+    {
+        StartUri = startUri;
+        return this;
+    }
+
+    /// <summary>
+    /// Enables the <see href="https://www.zaproxy.org/docs/desktop/addons/ajax-spider/automation/">ZAP Ajax
+    /// Spider</see>. This is useful if you have an SPA; it unnecessarily slows down the scan otherwise.
+    /// </summary>
+    public SecurityScanConfiguration UseAjaxSpider()
+    {
+        AjaxSpiderIsUsed = true;
+        return this;
+    }
+
+    /// <summary>
+    /// Signs in directly (see <see cref="AccountController.SignInDirectly(string)"/>) with the given user at the start
+    /// of the scan.
+    /// </summary>
+    /// <param name="userName">The name of the user to sign in with directly.</param>
+    public SecurityScanConfiguration SignIn(string userName = DefaultUser.UserName)
+    {
+        SignInUserName = userName;
+        return this;
+    }
+
+    /// <summary>
+    /// Excludes a given URL from the scan completely.
+    /// </summary>
+    /// <param name="excludedUrlRegex">
+    /// The regex pattern to match URLs against to exclude them. These should be patterns that match the whole absolute
+    /// URL, so something like ".*blog.*" to match /blog, /blog/my-post, etc.
+    /// </param>
+    public SecurityScanConfiguration ExcludeUrlWithRegex(string excludedUrlRegex)
+    {
+        ExcludedUrlRegexPatterns.Add(excludedUrlRegex);
+        return this;
+    }
+
+    /// <summary>
+    /// Disable a certain active scan rule for the whole scan. If you only want to disable a rule for a given page, use
+    /// <see cref="ExcludeUrlWithRegex(string)"/> instead.
```

Comment:
Because `ExcludeUrlWithRegex(string)` disables the scan altogether for the URLs matched.
```suggestion
    /// Disable a certain active scan rule for the whole scan. If you only want to disable a rule for specific pages
    /// matched by a regex, use <see cref="DisableScanRuleForUrlWithRegex(string, int)"/> instead.
```